pandas/io/excel/_base.py

```diff
@@ -809,17 +819,24 @@ class ExcelFile:
         "pyxlsb": _PyxlsbReader,
     }
 
-    def __init__(self, io, engine=None):
+    def __init__(self, path_or_io, engine=None):
         if engine is None:
             engine = "xlrd"
+            if isinstance(path_or_io, IOBase):
+                if _is_ods_stream(path_or_io):
+                    engine = "odf"
+            else:
+                ext = os.path.splitext(str(path_or_io))[-1]
+                if ext == ".ods":
```

Comment:
Rather than this should register the writer globally

https://github.com/pandas-dev/pandas/blob/120e9d9fae447be2e1cea3fd83a1f19501972cca/pandas/core/config_init.py#L576

Reply:
The weird thing is that this stuff does not seem to be used anywhere. Correct me if I'm wrong. I added a similar bit for the OpenOffice file format, but it did not seem to be called/tested anywhere.